specs/interop/token-bridging.md

```diff
@@ -72,26 +101,200 @@ contract OptimismSuperchainERC20 is ERC20, ISuperchainERC20 {
+- Current function names are open to discussion.
+
+## Implementation
+
+### Cross Chain transfers
+
+The standard will include the following three external functions:
+
+- `bridgeERC20(uint256 amount, uint256 chainId)`
+- `bridgeERC20To(address to, uint256 amount, uint256 chainId)`
+- `finalizeBridgeERC20(address to, uint256 amount)`
+
+The first two functions will burn `amount` tokens and initialize a message to the `L2ToL2CrossChainMessenger` to mint the `amount` in the destination (`to` address in `chainId`).
+
+`finalizeBridgeERC20(address to, uint256 amount)` will process incoming messages from the `L2ToL2CrossChainMessenger` initiated from the same token on a different `chainId` and mint `amount` to the `to` address.
+
+A possible implementation would look like this:
+
+```solidity
+function bridgeERC20(uint256 _amount, uint256 _chainId) external returns (bool success) {
+    success = bridgeERC20To(msg.sender, _amount, _chainId);
+  }
+
+function bridgeERC20To(address _to, uint256 _amount, uint256 _chainId) public returns (bool) {
+  _burn(msg.sender, _amount);
+  bytes memory _message = abi.encodeCall(this.finalizeBridgeERC20, (_to, _amount));
+  _sendMessage(_chainId, _message);
+  return true;
+}
+
+function finalizeBridgeERC20(address _to, uint256 _amount) external onlyMessenger {
+  _mint(_to, _amount);
+}
+
+/////// Internals and modifiers ///////////
+function _sendMessage(uint256 _destination, bytes memory _data) internal virtual {
+  L2ToL2CrossDomainMessenger.sendMessage(_destination, address(this), _data);
+}
+
+modifier onlyMessenger() virtual {
+  require(msg.sender == address(L2ToL2CrossChainMessenger));
+  require(L2ToL2CrossChainMessenger.crossDomainMessageSender() == address(this));
+  _;
+}
+```
+
+Note: `send()` or `xTransfer()` can also be good fits instead of `bridgeERC20()`.
+
+### Cross Chain `transferFrom`
+
+The standard will not include a remotely initialized `transfer` implementation as we think this should be handled directly on the chain where the funds live with the `bridgeERC20()` function. `transferFrom`, on the other hand, might get initialized from a call to a contract that is localized in a different chain from the funds. To be composable, the standard will introduce a `remoteTransferFrom()` function.
+
+Notice that this action is no longer atomic, so the contract will need a special implementation to use it and resume execution once the funds arrive in the chain.
+
+**Approvals**
+
+To enable `remoteTransferFrom()`, we must include a separate `xAllowance` mapping that keeps track of the `chainId`. This allowance should be set locally using a new `xApprove()` function (the naming is open for discussion).
+
+Here is what a possible implementation could look like:
+
+```solidity
+function xApprove(address _spender, uint256 _chainId, uint256 _amount) external returns (bool) {
+  // TODO [research] _chainId wildcard as type(uint256).max (this is very risky)
```

Comment:
I think we should leave out the wildcard because that would conflict where the chainid is `type(uint256).max`. A future `permit3` contract could exist in the future and add that functionality